src/Microsoft.OData.Core/UriParser/Resolver/ODataUriResolver.cs

```diff
@@ -530,5 +544,29 @@ internal static ODataUriResolver GetUriResolver(IServiceProvider container)
                 }
             }
         }
+
+        private static CaseInsensitiveSchemaElementsCache GetCaseInsensitiveSchemaElementsCache(IEdmModel model)
+        {
+            if (model == null)

```

Comment:
I think a `Debug.Assert` should suffice here since this is a private method. The null check should be in public/internal methods or constructors